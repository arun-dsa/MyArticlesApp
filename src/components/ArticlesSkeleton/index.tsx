import {
  Card,
  CardActions,
  CardContent,
  Divider,
  Skeleton,
} from "@mui/material";

import { CardBodyText, CardTitleText } from "@components/StyledComponents";

const ArticlesSkeleton = ({ id }: { id: string | number }) => {
  return (
    <Card
      sx={{
        width: "100%",
        height: "100%",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <CardContent sx={{ flex: 1 }}>
        <CardTitleText
          gutterBottom
          variant="h5"
          data-testid={`article-title-skeleton${id}`}
        >
          <Skeleton width="100%" height={45} />
        </CardTitleText>
        <CardBodyText
          variant="body2"
          color="text.secondary"
          data-testid={`article-summary-skeleton${id}`}
        >
          <Skeleton width="100%" height={70} />
        </CardBodyText>
      </CardContent>
      <Divider />
      <CardActions sx={{ justifyContent: "flex-end" }}>
        <Skeleton
          width={100}
          height={40}
          data-testid={`article-share-skeleton${id}`}
        />
        <Skeleton
          width={150}
          height={40}
          data-testid={`article-view-full-skeleton${id}`}
        />
      </CardActions>
    </Card>
  );
};

export default ArticlesSkeleton;
