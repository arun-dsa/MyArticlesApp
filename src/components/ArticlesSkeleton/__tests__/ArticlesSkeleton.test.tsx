import { ReactNode } from "react";
import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import { routeWrapper } from "@utils/test_utils";
import ArticlesSkeleton from "..";

describe("Article card skeleton test", () => {
  test("Test render article card skeleton", () => {
    const ArticleRoute: ReactNode = routeWrapper([
      { path: "/", element: <ArticlesSkeleton id="1" /> },
    ]);
    const container = render(ArticleRoute);
    expect(container).toMatchSnapshot();
  });
});
