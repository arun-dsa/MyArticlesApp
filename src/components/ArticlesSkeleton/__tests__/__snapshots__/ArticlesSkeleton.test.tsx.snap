// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Article card skeleton test > Test render article card skeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-97bign-MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
            data-testid="article-title-skeleton1"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 100%; height: 45px;"
            />
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
            data-testid="article-summary-skeleton1"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
              style="width: 100%; height: 70px;"
            />
          </p>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
            data-testid="article-share-skeleton1"
            style="width: 100px; height: 40px;"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
            data-testid="article-view-full-skeleton1"
            style="width: 150px; height: 40px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1rp1zv7-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-97bign-MuiCardContent-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1u4kd2-MuiTypography-root"
          data-testid="article-title-skeleton1"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
            style="width: 100%; height: 45px;"
          />
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1i87a3m-MuiTypography-root"
          data-testid="article-summary-skeleton1"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
            style="width: 100%; height: 70px;"
          />
        </p>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1mgzbsd-MuiCardActions-root"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
          data-testid="article-share-skeleton1"
          style="width: 100px; height: 40px;"
        />
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1e1x4vv-MuiSkeleton-root"
          data-testid="article-view-full-skeleton1"
          style="width: 150px; height: 40px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
